core/mm/core_mmu.c

```diff
@@ -681,6 +681,12 @@ static void verify_special_mem_areas(struct memory_map *mem_map,
 	 */
 	for (mem = start; mem < end; mem++) {
 		for (n = 0; n < mem_map->count; n++) {
+			/*
+			 * Ignore MEM_AREA_SEC_RAM_OVERALL since it can
+			 * cover all secure memory, even if it's special.
+			 */
+			if (mem_map->map[n].type == MEM_AREA_SEC_RAM_OVERALL)
+				continue;
```

Comment:
This is applicable when SDP_TEST_MEM that is inside the SEC_RAM_OVERALL. But not if the SDP memory is defined with `CFG_TEE_SDP_MEM_BASE/_SIZE`. Maybe specifically test here if  `mem->start == TEE_SDP_TEST_MEM_BASE && mem->size == TEE_SDP_TEST_MEM_SIZE`.